src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/Offender.java

```diff
@@ -193,4 +197,106 @@ public OffenderIdentifier addIdentifier(final String type, final String value) {
         identifiers.add(offenderIdentifier);
         return offenderIdentifier;
     }
+
+    public List<ExternalMovement> getAllMovements() {
+        final var externalMovements = new ArrayList<ExternalMovement>();
+        bookings.forEach(b -> externalMovements.addAll(b.getExternalMovements()));
+        return externalMovements.stream()
+            .sorted(Comparator.comparing(ExternalMovement::getMovementTime))
+            .collect(Collectors.toList());
+    }
+
+    public void addBooking(final OffenderBooking booking) {
+        booking.setOffender(this);
+        bookings.add(booking);
+        bookings.forEach(OffenderBooking::incBookingSequence);
+
+    }
+    public PrisonerInPrisonSummary getPrisonerInPrisonSummary() {
+        final var movementMap = getAllMovements()
+            .stream()
+            .filter(f -> f.getMovementType() != null && List.of("ADM", "REL", "TAP").contains(f.getMovementType().getCode()))
+            .collect(Collectors.groupingBy(ExternalMovement::getOffenderBooking));
+
+        final var summary = PrisonerInPrisonSummary.builder()
+            .prisonerNumber(getNomsId())
+            .prisonPeriod(movementMap.entrySet().stream()
+                .filter(p -> p.getValue().size() > 0)
+                .map(e ->
+                    PrisonPeriod.builder()
+                        .bookingId(e.getKey().getBookingId())
+                        .bookNumber(e.getKey().getBookNumber())
+                        .movementDates(buildMovements(e.getValue()))
+                        .build())
+                .collect(toList())
+            )
+            .build();
+
+        summary.getPrisonPeriod().forEach(
+            pp -> {
+                pp.setEntryDate(pp.getMovementDates().stream().findFirst()
+                    .filter( m -> "ADM".equals(m.getInwardType()) )
+                    .map(MovementDate::getDateInToPrison).orElse(null));
+
+                pp.setReleaseDate(pp.getLastMovement()
+                    .filter( m -> "REL".equals(m.getOutwardType()) )
+                    .map(MovementDate::getDateOutOfPrison).orElse(null));
+            }
+        );
```

Comment:
We could do a test case which is 
ADMISSION to Prison A
TAP OUT
TRANSFER IN FROM TAP Prison B

That is a valid movement scenario I think